### PR TITLE
Add `--user` flag to global dvc auto installation

### PR DESCRIPTION
**Pull request by julieg18**
* uses the python extension api to get information on selected environment and, if env is global, adds a `--user` flag to the install command 

Part of #3935

### CODE REVIEW

File: extension/src/extensions/python.ts [hunk 1]
```diff
@@ -11,15 +11,24 @@ interface Settings {
   }
 }
 
-type EnvironmentVariables = { readonly [key: string]: string | undefined }
+type EnvironmentVariables = { readonly [key: string]: undefined }
```
Comment (mattseddon) [inline extension/src/extensions/python.ts:14]: [Q] is this change intentional?

File: extension/src/setup/autoInstall.ts [hunk 2]
```diff
@@ -64,7 +81,12 @@ const showInstallProgress = (
     try {
       await Toast.runCommandAndIncrementProgress(
         async () => {
-          await installPackages(root, pythonBinPath, 'dvc')
+          await installPackages(
```
Comment (mattseddon) [inline extension/src/setup/autoInstall.ts:84]: I think some duplication could be removed here: 

```
    try {
      await Toast.runCommandAndIncrementProgress(
        async () => {
          await installPackages(
            root,
            pythonBinPath,
            ...getProcessGlobalArgs(isGlobalEnv),
            'dvc'
          )
          return 'DVC Installed'
        },
        progress,
        75
      )
      return Toast.delayProgressClosing()
    } catch (error: unknown) {
      return Toast.reportProgressError(error, progress)
    }
```
(especially with the upgrade command being basically the same with a different return value)

File: extension/src/extensions/python.ts [hunk 3]
```diff
@@ -11,15 +11,24 @@ interface Settings {
...
+    type: string
+  }
+}
+
 export interface VscodePython {
   ready: Thenable<void>
   settings: Settings
   environments: {
+    known: Environment[]
+    getActiveEnvironmentPath: () => { id: string }
```
Comment (mattseddon) [inline extension/src/extensions/python.ts:31]: [I] If we are using this we also need to use `onDidChangeActiveEnvironmentPath` so that updates are taken into account... unless an update will be triggered in a different way.

[Q] Can we replace our other use of the API with this now?

**Review by mattseddon — APPROVED** [review]

File: extension/src/extensions/python.ts (hunk 3 above)
Comment (julieg18) [reply extension/src/extensions/python.ts:31]: > [I] If we are using this we also need to use onDidChangeActiveEnvironmentPath so that updates are taken into account... unless an update will be triggered in a different way.

Updates are triggered with `onDidChangePythonExecutionDetails` since that runs when the environment path changes. 

> [Q] Can we replace our other use of the API with this now?

Apologies, I'm not sure what you mean. What other use of the API could be replaced? Looking at the API, they all seem to be doing different things to me 🤔

File: extension/src/setup/autoInstall.ts (hunk 2 above)
Comment (julieg18) [reply extension/src/setup/autoInstall.ts]: Cleaned up some of the duplication!

File: extension/src/extensions/python.ts (hunk 3 above)
Comment (julieg18) [reply extension/src/extensions/python.ts:31]: Going to merge this, but happy to update the use of the API in a followup!

Comment (mattseddon) [reply extension/src/extensions/python.ts:31]: We use `getExecutionDetails` to get the Python executable for the active environment. Isn't that returned by `getActiveEnvironmentPath`?

```
export type EnvironmentPath = {
    /**
     * The ID of the environment.
     */
    readonly id: string;
    /**
     * Path to environment folder or path to python executable that uniquely identifies an environment. Environments
     * lacking a python executable are identified by environment folder paths, whereas other envs can be identified
     * using python executable path.
     */
    readonly path: string;
};
```

☝🏻 maybe we can't rely on this as it could return a folder and not an executable. Nevermind.
